rust/private/rustc.bzl

```diff
@@ -1146,16 +1262,21 @@ def _make_link_flags_default(linker_input_and_use_pic):
                 "link-arg=-Wl,--no-whole-archive",
             ])
         else:
-            ret.extend(_portable_link_flags(lib, use_pic))
+            ret.extend(_portable_link_flags(lib, use_pic, ambiguous_libs))
     return ret
 
-def _add_native_link_flags(args, dep_info, linkstamp_outs, crate_type, toolchain, cc_toolchain, feature_configuration):
+def _libraries_dirnames(linker_input_and_use_pic_and_ambiguous_libs):
+    link_input, use_pic, _ = linker_input_and_use_pic_and_ambiguous_libs
+    return [get_preferred_artifact(lib, use_pic).dirname for lib in link_input.libraries]
```

Comment:
Nit: we could deduplicate, the current code will list `_ambiguous_libs` once per ambiguous library. Not sure this is at all necessary, it should be just a temporary solution, but..

Reply:
done